### PR TITLE
Add pre-commit configuration with black and run in GHA as listing

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

**Comment by blowekamp** [comment]
@jcfr I added pre-commit and black here. You mentioned[ blacked-docs](https://pypi.org/project/blacken-docs/0.2.0/). I have not used that before from the description it seems for just code inside python comments, so I'm not sure it needed here.

**Review by dave3d — APPROVED** [review]

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+max-line-length = 88
+extend-ignore = E203, W503
+max_complexity = 32
```
Comment (jcfr) [inline .flake8:4]: Setting should be added to `pyproject.toml`

Comment (jcfr) [reply .flake8:4]: See https://pypi.org/project/Flake8-pyproject/

Comment (blowekamp) [reply .flake8:4]: As of 6.0 this is not a documented configuration location for fake8: 
https://flake8.pycqa.org/en/6.0.0/user/configuration.html

Is there some more updated documentation of the feature of using the file for configuration?

**Comment by jcfr** [comment]
For new project I also suggest integrating `ruff` instead. I can provide an example of config or due a follow up PR.

**Pull request merged**